### DISCUSSION TITLE
ImagePolicy must be in the flux-system namespace?

### DISCUSSION THREAD

**Discussion by sagikazarmark** [fluxcd/flux2#4173] [Q&A] [answered]
I'm trying to wrap my head around how image automations work. Specifically, why the `ImagePolicy` objects have to be in the `flux-system` namespace (or rather - but this is just a theory - in the same namespace as the `ImageUpdateAutomation` object).

Every example and guide creates those resources in the `flux-system` namespace, but I couldn't find any explanation. (I do recall reading about this though, so I might have not found it)

In my case, I'd like to install multiple instances of the same application, but apply different image policies to them. I tried to create `ImagePolicy` objects in the dedicated namespace of each application instance. The relevant controller actually finds the tags, but nothing gets updated and I couldn't find any error.

If my above theory is correct, I could create an `ImageUpdateAutomation` object in each namespace (I don't have cross-namespace references disabled) which should potentially solve the problem (haven't tried it yet), but I'd like the automation to work the same way for each instance.

I find this behavior a bit weird, especially because there is no obvious failure, but there is a chance I just need to be educated. In any case, having to put `ImagePolicy` objects clearly tied to a namespace in `flux-system` feels weird.

**Comment by makkes** [comment, marked as answer]
The [Image Update Automations](https://fluxcd.io/flux/components/image/imageupdateautomations/) documentation has this in the very first sentence:

> The ImageUpdateAutomation type defines an automation process that will update a git repository, based on image policy objects in the same namespace.

So, yes, your theory is correct that ImageUpdateAutomations only apply ImagePolicies from the same Namespace.

> having to put ImagePolicy objects clearly tied to a namespace in flux-system feels weird.

This is not the case, see above. You can create Flux objects in any namespace you like.

If you'd like to deploy your app into multiple Namespaces and only maintain a single ImageUpdateAutomation for updating all of them at the same time, you can create Kustomizations for each app instance in Namespace "foo", set their respective `.spec.targetNamespace` to the Namespace you'd like to deploy that instance into and create an ImagePolicy and ImageUpdateAutomation in the same Namespaces as all the Kustomizations.

**Reply by sagikazarmark** [reply]
@makkes thanks for the clarification. I guess I missed that very first sentence.

Since the documentation and every example installs image policies into the flux-system namespace, I think it's easy to come to the conclusion that it's either required or some sort of best practice.

I don't yet know how, but I think it would make sense to spell this out for people like me who tends to miss the first (and most important) line of the documentation.

Thanks again!
